File: app/src/main/java/com/waygo/data/provider/interfaces/ILogBoxProvider.java
package com.waygo.data.provider.interfaces;


import android.support.annotation.NonNull;

import com.waygo.data.model.GeoCoordinate;
import com.waygo.data.model.fuel.Fuel;
import com.waygo.utilskt.Result;

import rx.Observable;

public interface ILogBoxProvider {

    @NonNull
    Observable<Result<Fuel>>  getTankLevel();

    @NonNull
    Observable<GeoCoordinate> getGeoPosition();
}

File: app/src/main/java/com/waygo/data/provider/interfaces/ILogBoxProvider.kt
package com.waygo.data.provider.interfaces


import com.waygo.data.model.GeoCoordinate
import com.waygo.data.model.fuel.Fuel
import com.waygo.utilskt.Result

import rx.Observable

public interface ILogBoxProvider {
    public fun getTankLevel(): Observable<Result<Fuel>>
    public fun getGeoPosition(): Observable<GeoCoordinate>
}
